num_enum_derive/src/lib.rs

```diff
@@ -278,6 +288,9 @@ impl Parse for EnumInfo {
             let mut has_default_variant: bool = false;
             let mut has_catch_all_variant: bool = false;
 
+            // Vec to keep track of the used discriminants and alt values.
+            let mut val_set: Vec<i128> = vec![];
```

Comment:
Yes there is a reason! 

I started originally to use a `BtreeSet`, but when I was going trought the code and throught the repo past issues. I found out issue #18 and read about `no_std`. I don't have any experience with embedded development (yet) but I understood the need and requirement. To my understanding, `BtreeSet` requires `std` because of hashing (`Hash` trait) while `Vec` requires only `alloc`. Also there are already usage of `Vec` in the code so I figured that I would not break anything with it.

 While `BtreeSet` is more natural, `Vec` keeps the repo `no_std` (and also as a small bonus, I guess it is faster as well since we don't do any hashing and the sorting is cheaper I think).

I might be also plain wrong and using a `BtreeSet` in a proc-macro is fine even in a `no_std` setting since it's used only for code generation. Let me know if I understood correctly.

Reply:
@illicitonion, I will resolve this only with your approval and feedback.

Reply:
I see! Thank you for being so diligent and looking through the history of the crate and its development!

This is one of the weird aspects of proc macros - the proc macro code itself can use all of the features of std it wants (similar to how the rust compiler itself can use any features it wants), the crucial thing is that the code being _generated_ needs to not use std features in order to be std-compliant. There's also an interesting grey-area where we expose some of our own code that gets depended on (e.g. https://github.com/illicitonion/num_enum/blob/main/num_enum/src/lib.rs), but essentially, any code in `num_enum_derive` which we run can use whatever it wants.

So it's ok to use a `BTreeSet` in our code, as long as we don't end up generating a `BTreeSet` in any generated code.

If you can switch this over to use a `BTreeSet`, I think we're ready to merge this change - thanks again for taking the time!

Reply:
Thank you for the explaination, now I know.

I changed it to BTreeSet.
And add the behavior discussed in #88